packages/chain-controller/src/ChainController.ts

```diff
@@ -0,0 +1,160 @@
+import type {
+  ControllerGetStateAction,
+  ControllerStateChangeEvent,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import { BaseController } from '@metamask/base-controller';
+import type { CaipAssetType, BalancesResult, Chain } from '@metamask/chain-api';
+import type { InternalAccount } from '@metamask/keyring-api';
+import type { HandleSnapRequest as SnapControllerHandleSnapRequestAction } from '@metamask/snaps-controllers';
+import type { SnapId } from '@metamask/snaps-sdk';
+import type { CaipChainId, Json } from '@metamask/utils';
+
+import { SnapChainProviderClient } from './SnapChainProviderClient';
+import { SnapHandlerClient } from './SnapHandlerClient';
+
+const controllerName = 'ChainController';
+
+export type ChainControllerState = Record<string, Json>;
```

Comment:
Two things:

1. Good catch, I started using `Record` to make the type system happy, my initial ideal was to use `{}` which seems to be forbidden

2. For now, no, we have no "persisted" state, we do have a "in-memory" one though, for the providers. So ultimately, this could be a plain class as-is, yes. In the future we might need a persistent state yes.

So what do you recommend/prefer here?
- I can make it a "look-alike" controller (like the messenger controller), and we could derive from the `BaseController` later on only if we need a state?
- Or I keep it as is (given that we might have a persistent state "soon")?

Reply:
Also, if we keep it as is, I could go for `Record<string, never>` instead, so this way this is similar to `{}` (since no other construction will match this type). WDYT?

Reply:
Okay, good questions. I guess if we plan on having persisted state in the future, we can keep it as a controller and use `Record<string, never>` for the state.